src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp

```diff
@@ -1186,37 +1186,67 @@ int xcldev::device::runTestCase(const std::string& py,
     std::string devInfoPath = name + "/test/";
     std::string xsaXclbinPath = getXsaPath(vendor) + devInfoPath;
     std::string dsaXclbinPath = dsaPath + devInfoPath;
-    std::string xrtTestCasePath = xrtPath + "test/" + py;
+    std::string xrtTestCasePath = xrtPath + "test/";
 
     output.clear();
 
     std::string xclbinPath;
     searchXsaAndDsa(m_idx, xsaXclbinPath, dsaXclbinPath, xclbinPath, output);
-    xclbinPath += xclbin;
-
-    if (stat(xrtTestCasePath.c_str(), &st) != 0 || stat(xclbinPath.c_str(), &st) != 0) {
-        //if bandwidth xclbin isn't present, skip the test
-        if(xclbin.compare("bandwidth.xclbin") == 0) {
-            output += "Bandwidth xclbin not available. Skipping validation.";
-            return -EOPNOTSUPP;
-        }
-        output += "ERROR: Failed to find ";
-        output += py;
-        output += " or ";
-        output += xclbin;
-        output += ", Shell package not installed properly.";
-        return -ENOENT;
-    }
+    std::string cmd;
+    
+    // if platform.json exists in the new shell pakcage then use the new testcase
+    // else fallback to old python test cases
+    
+    // NEW FLOW: runs if platform.json is available and the platform is not versal
+    // Note: we don't have cpp based versal bw testcase
```

Comment:
I don't think you want to call out versal. there is no difference than other platforms here.